drv/ImmutableList.drv

```diff
@@ -436,5 +481,378 @@ public class IMMUTABLE_LIST KEY_GENERIC extends ABSTRACT_LIST KEY_GENERIC implem
 	}
 #endif
 
+	// Overridden mutation methods to make unsupported.
+
+	/**
+	 * Always throws {@link UnsupportedOperationException} as this is an immutable type.
+	 *
+	 * @deprecated
+	 */
+	@Override
+	@Deprecated
+	public final void add(final int index, final KEY_GENERIC_TYPE k) {
+		throw new UnsupportedOperationException();
+	}
+
+	/**
```

Comment:
I think these should be all {@inheritDoc} at the start and have @implSpec tagging the implementation specification.